src/pages/sa-trx-dtd/form/DocumentTypeDefaultForm.js

```diff
@@ -0,0 +1,250 @@
+import { Checkbox, FormControlLabel, Grid } from '@mui/material'
+import { useContext, useEffect } from 'react'
+import * as yup from 'yup'
+import FormShell from 'src/components/Shared/FormShell'
+import toast from 'react-hot-toast'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import { useInvalidate } from 'src/hooks/resource'
+import { ResourceIds } from 'src/resources/ResourceIds'
+import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
+import { Grow } from 'src/components/Shared/Layouts/Grow'
+import { useForm } from 'src/hooks/form'
+import { SystemRepository } from 'src/repositories/SystemRepository'
+import ResourceComboBox from 'src/components/Shared/ResourceComboBox'
+
+import { ControlContext } from 'src/providers/ControlContext'
+import { SaleRepository } from 'src/repositories/SaleRepository'
+import { InventoryRepository } from 'src/repositories/InventoryRepository'
+import { DataSets } from 'src/resources/DataSets'
+
+export default function DocumentTypeDefaultForm({ labels, maxAccess, recordId, dtId, functionId }) {
+  const { platformLabels } = useContext(ControlContext)
+
+  const { getRequest, postRequest } = useContext(RequestsContext)
+
+  const invalidate = useInvalidate({
+    endpointId: SaleRepository.DocumentTypeDefault.qry
+  })
+
+  const { formik } = useForm({
+    initialValues: {
+      recordId: dtId || null,
+      commitItems: false,
+      dtId: '',
+      plantId: '',
+      spId: '',
+      allocateBy: null,
+      postMetalToFinancials: false,
+      siteId: '',
+      disableSKULookup: false
+    },
+    maxAccess,
+    enableReinitialize: false,
+    validationSchema: yup.object({
+      dtId: yup.string().required(),
+      allocateBy: yup
+        .string()
+        .transform(value => (value === null ? '' : value))
```

Comment:
error in edit mode if clear value of allocateBy field , field coloris not change to red